### PR TITLE
Fix failure to find mednafen.cfg when inside bin/ or lib/ folders on Windows

### LINKED ISSUE #129: Cannot find mednafen.cfg inside bin/ or lib/ folders on Windows
If mednaffe.exe is located in a directory whose path contains "bin" or "lib", mednafen.cfg cannot be found (even if in the same folder as mednaffe.exe).

For eg.
if the path for mednaffe.exe is D:\bin\mednaffe\mednaffe.exe
and the path for mednafen.cfg is D:\bin\mednaffe\mednafen.cfg,
(same directory)
mednafen.cfg will still not be found. This is true for "lib" as well.

### PR DESCRIPTION
**Pull request by mayawarrier**
It seems from reading the code that the intent is to look in the process directory for mednafen.cfg.
`g_win32_get_package_installation_directory_of_module()` returns the parent directory when the path contains "bin" or "lib".

I switched out 
`g_win32_get_package_installation_directory_of_module(NULL)`
with new function 
`win32_get_process_directory()`
instead.

Resolves #129

### CODE REVIEW

**Comment by AmatCoder** [comment]
Thanks for patch, it is correct but it has two issues:

1. Building Mednaffe with Cygwin is not supported.
2. `g_win32_get_package_installation_directory_of_module` function is used in [more places](https://github.com/AmatCoder/mednaffe/search?q=g_win32_get_package_installation_directory_of_module).

**Comment by mayawarrier** [comment]
@AmatCoder Just made the required fixes. Let me know if more changes are needed.

File: src/win32util.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * along with Mednaffe; if not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ */
+
+
+#ifndef __WIN32UTIL_H__
+#define __WIN32UTIL_H__
+
+#include <gtk/gtk.h>
```
Comment (AmatCoder) [inline src/win32util.h:28]: `#include <glib.h>` would be better.

File: src/win32util.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with Mednaffe; if not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ */
+
+
+#include "win32util.h"
+
+#ifdef G_OS_WIN32
```
Comment (AmatCoder) [inline src/win32util.c:27]: No need for `#ifdef G_OS_WIN32`. This is already Windows-specific code.

File: src/win32util.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+#ifdef G_OS_WIN32
```
Comment (AmatCoder) [inline src/win32util.h:30]: No need for `#ifdef G_OS_WIN32`. This is already Windows-specific code.

**Review by AmatCoder — CHANGES_REQUESTED** [review]
Also `win32util.h` and `win32util.c` files should be added to `mednaffe_SOURCES` into [Makefile.am](https://github.com/AmatCoder/mednaffe/blob/master/src/Makefile.am#L43).

**Comment by mayawarrier** [comment]
hey @AmatCoder, just following up on this PR :)

File: src/Makefile.am [hunk 4]
```diff
@@ -41,6 +41,7 @@ if LINUX
 else
 if WINDOWS
   mednaffe_SOURCES+= widgets/joystick_windows.c
+  mednaffe_SOURCES+= win32util.c win32util.h
```
Comment (AmatCoder) [inline src/Makefile.am:44]: Here you need add the files to the previous one:
`mednaffe_SOURCES+= win32util.c win32util.h widgets/joystick_windows.c`

**Review by AmatCoder — CHANGES_REQUESTED** [review]

**Review by AmatCoder — APPROVED** [review]

**Comment by AmatCoder** [comment]
Merged. Thanks very much!

**Pull request merged**